backend/b2: Switch to official library `Backblaze/blazer`

adamantike opened the pull request:
What does this PR change? What problem does it solve?
-----------------------------------------------------

According to https://github.com/kurin/blazer Readme, the repository is no longer maintained, and the official one at
https://github.com/Backblaze/blazer took over.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

Closes #4148

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

MichaelEischer commented on go.mod:
```diff
@@ -5,6 +5,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.0
 	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.1.0
+	github.com/Backblaze/blazer v0.6.1
```
Please revert all changes to go.mod _except_ for the blazer library update. Updating the other library belongs into a separate PR.

adamantike replied:
Is this desired? The changes I included here were just the results of a `go mod tidy`. The other library updates are because of `blazer` now including a [`go.mod` file](https://github.com/Backblaze/blazer/blob/efa10c0fffe98ce85c0a927a13249cc695a4f8e5/go.mod), and requiring those newer versions.

MichaelEischer replied:
Oh indeed you're right. I'd have prefer to update those dependencies separately, but we can also update them right now.

MichaelEischer reviewed (changes requested):
See above

MichaelEischer reviewed (approved):
LGTM.

MichaelEischer committed:
backend/b2: Switch to official library `Backblaze/blazer`

According to https://github.com/kurin/blazer Readme, the repository is
no longer maintained, and the official one at
https://github.com/Backblaze/blazer took over.

Closes #4148

MichaelEischer commented:
(rebased to fix the merge conflict)
